python/pyspark/sql/connect/udf.py

```diff
@@ -99,9 +97,7 @@ def __init__(
             )
 
         self.func = func
-        self._returnType = (
-            parse_data_type(returnType) if isinstance(returnType, str) else returnType
-        )
+        self._returnType = returnType
```

Comment:
In the vanilla PySpark, a `returnType` property is defined as https://github.com/apache/spark/blob/master/python/pyspark/sql/udf.py#L233, which returns DataType always.

In the existing Connect's code, `wrapper.returnType = self._returnType` is utilized instead.

I am afraid that line of change may break that property parity.